r-python-eds-lessons/Dockerfile

```diff
@@ -0,0 +1,27 @@
+FROM continuumio/anaconda3
+
+RUN apt-get -y install r-base-core && \
```

Comment:
This installation step can be improved by: 

1. Running apt-get update
2. Adding the `--no-install-recommends` tag to `apt-get install`
3. Providing the list of packages in alphabetical order to one `install` call instead of calling `install` once for each package, e.g., 

```bash

RUN apt-get update \
  && apt-get install -y --no-install-recommends \
    libcurl4-openssl-dev \
    libgeos-dev \
    libgdal-dev \
    libproj-dev \
    libssl-dev \
    libudunits2-dev \
    r-base-core

```

Reply:
@mbjoseph I noticed the build is now failing here: 

https://github.com/jvtcl/dockerfiles/blob/2b2e1c3f295d16595da3a38664cee054d025cc53/r-python-eds-lessons/Dockerfile#L15

I'm seeing this error for some packages:
 
```
sh: 1: make: not found
ERROR: compilation failed for package ‘sys’
```

And this one in others: 

<img width="843" alt="Screen Shot 2019-06-05 at 11 58 12 AM" src="https://user-images.githubusercontent.com/10080304/58978519-36225600-8789-11e9-8c12-b3119ebd9686.png">

Looks like a compilation error, but I'm not sure what might be triggering it.

Reply:
You may have to install the `build-essential` package as well: 

```
RUN apt-get update \
  && apt-get install -y --no-install-recommends \
    build-essential \
    libcurl4-openssl-dev \
    libgeos-dev \
    libgdal-dev \
    libproj-dev \
    libssl-dev \
    libudunits2-dev \
    r-base-core
```

If you want, you could remove the `--no-install-recommends` flag, but it will most likely result in an image that is larger than it needs to be.

Reply:
@mbjoseph the `--no-install-recommends` flag does appear to be causing problems, even with `build-essential` in place. 

I noticed that [`nlme` depends on R version `3.4` and above](https://cran.r-project.org/web/packages/nlme/index.html), yet version `3.3.3` installs in the build. The error returned when `--no-install-recommends` is present is related to this: 

```
Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)
Warning message:
package ‘nlme’ is not available (for R version 3.3.3) 
Error: Installation of package(s) nlme was unsuccessful.
Execution halted
```
Yet when that flag is left out, the build succeeds on the same R version.

Reply:
Ok - if it's going to be an issue, then feel free to remove `--no-install-recommends`!